dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPIImpl.java

```diff
@@ -342,25 +367,33 @@ private IHTMLPage getPageByUri(final PageMode mode, final Host host, final Strin
         return htmlPage;
     }
 
+    private IHTMLPage getPageById(final PageMode mode, final Host host, final String id)
+            throws DotDataException, DotSecurityException {
+
+        final HttpServletRequest request = HttpServletRequestThreadLocal.INSTANCE.getRequest();
+        final Language language = this.getCurrentLanguage(request);
+
+        return APILocator.getHTMLPageAssetAPI().findByIdLanguageFallback(id, language.getId(), mode.showLive, userAPI.getSystemUser(),
+                mode.respectAnonPerms);
+    }
+
     private Language getCurrentLanguage(final HttpServletRequest request) {
         return request != null ? this.languageWebAPI.getLanguage(request) : this.languageAPI.getDefaultLanguage();
     }
 
     public class HTMLPageUrl {
-        private String pageUrl;
         private String pageUrlMapper;
         private HTMLPageAsset htmlPage;
         private Boolean hasLive = null;
 
-        public HTMLPageUrl(final String pageUrl, final String pageUrlMapper, final Boolean hasLive) {
-            this.pageUrl = pageUrl;
+        public HTMLPageUrl(final HTMLPageAsset htmlPage, final String pageUrlMapper, final Boolean hasLive) {
```

Comment:
The constructor signature was modified here too

Reply:
agree

Reply:
this a internal class, however I am going to change the method scope to private